python/rascaline/rascaline/utils/clebsch_gordan/_cg_cache.py

```diff
@@ -18,55 +20,174 @@
 except ImportError:
     HAS_MOPS = False
 
+
 try:
+    import torch
     from torch import Tensor as TorchTensor
+
+    torch_dtype = torch.dtype
+    torch_device = torch.device
+
+    HAS_TORCH = True
 except ImportError:
+    HAS_TORCH = False
 
     class TorchTensor:
         pass
 
+    class torch_dtype:
+        pass
+
+    class torch_device:
+        pass
+
 
 UNKNOWN_ARRAY_TYPE = (
     "unknown array type, only numpy arrays and torch tensors are supported"
 )
 
 
-# =================================
-# ===== ClebschGordanReal class
-# =================================
+def calculate_cg_coefficients(
+    lambda_max: int,
+    sparse: bool = True,
+    use_mops: Optional[bool] = None,
+    use_torch: bool = False,
+) -> TensorMap:
+    """
+    Calculates the Clebsch-Gordan coefficients for all possible combination of l1 and
+    l2, up to ``lambda_max``. Returns them in :py:class:`TensorMap` format.
 
+    The output data structure of the output :py:class:`TensorMap` depends on the backend
+    used to perform CG tensor products, currently: ["python-dense", "python-sparse",
+    "mops"].
 
-class ClebschGordanReal:
+    ``cg_backend="python-dense"``:
+        - samples: `sample`, i.e. a dummy sample.
```

Comment:
```suggestion
        - samples: `_`, i.e. a dummy sample.
```

We use `_` elsewhere for a single label entry with no associated meaning